### ISSUE TITLE
Extra Check - Quota check

### ISSUE THREAD

**Issue by JvanderMeer-GitHub** [ykuijs/SPPeriodicChecks#8]
I created the following to check if sites have a quota and to generate a certain warning if x% has been reached for the fill factor.

```PowerShell
function script:Check22_QuotaCheck() {
    $sb = [Scriptblock]::Create({
WriteLog "Starting Check 22: Quota check"
$results.Check22 = ""

$templates = [Microsoft.SharePoint.Administration.SPWebService]::ContentService.quotatemplates
$tFound = $false
$templateName = "No template found"
$siteFillPercentageCheck = 90


$webapplications = Get-SPWebApplication

if ($null -eq $webapplications)
{
    # No Web Applications found
    WriteLog "  Check not possible: No web applications found. Either because none exist or an access issue occurred."
    $results.Check22 = $results.Check22 + "Quota Check: No web applications found.`r`n"
}
else
{
    # One or more Web Applications found
    foreach ($webapplication in $webapplications)
    {
	$sites = Get-SPSite -Limit ALL -WebApplication $webapplication

		foreach ($site in $sites)  
		{
			$URL = $site.URL
			$site.RecalculateStorageMetrics()
			
			foreach ($qt in $templates)  
			{  
				if ($qt.QuotaId -eq $site.Quota.QuotaID)  
				{  
					   $templateName = $qt.Name;   
					   $tFound = $true  
				}
			}
				if ($tFound -eq $true)  
				{
					$siteFillPercentage = [math]::Round(($site.Usage.Storage/1MB)*100 / ($site.Quota.StorageMaximumLevel/1MB))
					
					if($siteFillPercentage -eq $siteFillPercentageCheck -OR $siteFillPercentage -gt $siteFillPercentageCheck)
					{
					WriteLog "  Check Failed"
					$results.Check22 = $results.Check22 + "`tSite filled for $siteFillPercentageCheck%: $URL)`r`n"
					}
				}
				if ($tFound -eq $false)  
				{
					$templateName = “No Template Applied”
					$StorageUsed = [math]::Round($site.Usage.Storage/1MB)
					WriteLog "  Check Failed"
					$results.Check22 = $results.Check22 + "`tNo Quota template for: $URL`r`n"
				}

			$tFound=$false
			$site.Dispose()   
		}  
    }
	
if ($results.Check22 -eq "")
{
WriteLog "  Check Passed"
$results.Check22 = $results.Check22 + "Quota Check Passed`r`n"
}
}

WriteLog "Completed Check 22: Quota Log check"
})

    return $sb.ToString()
}
```

**Comment by ykuijs** [comment]
Will add this to the list of checks

**Comment by ykuijs** [comment]
Added in next version

**Comment by ykuijs** [comment]
Fixed via PR #12
